influxdb_iox/src/commands/catalog/topic.rs

```diff
@@ -43,10 +44,23 @@ enum Command {
 pub async fn command(config: Config) -> Result<(), Error> {
     match config.command {
         Command::Update(update) => {
+            // for each topic name given, update the catalog, await and collect the response, print
+            // the IDs and return on error
             let catalog = update.catalog_dsn.get_catalog("cli").await?;
             let topics_repo = catalog.kafka_topics();
-            let topic = topics_repo.create_or_get(&update.db_name).await?;
-            println!("{}", topic.id);
+            update
+                .topic_names
+                .iter()
+                .map(|n| async move { topics_repo.create_or_get(n).await })
+                .collect::<FuturesUnordered<_>>()
+                .collect::<Vec<_>>()
+                .await
+                .into_iter()
+                .collect::<Result<Vec<_>, _>>()?
+                .iter()
+                .for_each(|t| {
+                    println!("{}", t.id);
+                });
```

Comment:
If you import [`TryStream`](https://docs.rs/futures/latest/futures/stream/trait.TryStreamExt.html#) it handles `Result` nicely - in this example, [`try_for_each`](https://docs.rs/futures/latest/futures/stream/trait.TryStreamExt.html#method.try_for_each) might be helpful!

```suggestion
            update
                .topic_names
                .iter()
                .map(|n| async move { topics_repo.create_or_get(n).await })
                .collect::<FuturesUnordered<_>>()
                .try_for_each(|t| std::future::ready(Ok(println!("{}", t.id))))
                .await?;
```

Reply:
much nicer :) thanks